swhkd/src/config.rs

```diff
@@ -272,11 +274,24 @@ pub struct Mode {
     pub name: String,
     pub hotkeys: Vec<Hotkey>,
     pub unbinds: Vec<KeyBinding>,
+    pub options: ModeOptions,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct ModeOptions {
+    pub swallow: bool,
+    pub onceoff: bool,
+}
+
+impl ModeOptions {
+    pub fn default() -> Self {
+        Self { swallow: false, onceoff: false }
+    }
```

Comment:
You could derive the Default trait directly here:
```rust
#[derive(Debug, Clone, PartialEq, Default)]
pub struct ModeOptions {
    pub swallow: bool,
    pub onceoff: bool,
}
```
See: https://doc.rust-lang.org/std/default/trait.Default.html

It will derive the same thing as your implementation, but I think it would be a bit nicer as the Default trait is a bit more generic.

Reply:
Thanks for this advice!